.github/workflows/release-build.yml

```
@@ -29,87 +31,40 @@ jobs:
         with:
           python-version: '3.10'
 
-      - name: Validate Input Version and Branch
-        id: validate_input_version
-        run: |
-          RELEASE_VERSION="${{ github.event.inputs.release_version }}"
-          echo "Using manually specified version: ${RELEASE_VERSION}"
-          MANUAL_CREATED_RELEASE_BRANCH="releases/${RELEASE_VERSION}"
-          echo "Expected manual created release branch: ${MANUAL_CREATED_RELEASE_BRANCH}"
-
-          # Fetch all remote heads and tags
-          git fetch --all --tags
-
-          # Check if the release branch exists remotely
-          if ! git ls-remote --heads origin ${MANUAL_CREATED_RELEASE_BRANCH} | grep -q refs/heads/${MANUAL_CREATED_RELEASE_BRANCH}; then
-            echo "Error: Manual release branch ${MANUAL_CREATED_RELEASE_BRANCH} does not exist remotely."
-            exit 1
-          else
-            echo "Found manual created release branch ${MANUAL_CREATED_RELEASE_BRANCH} remotely."
-          fi
-
-          echo "manual_created_release_branch=${MANUAL_CREATED_RELEASE_BRANCH}" >> $GITHUB_OUTPUT
+      - name: Install the latest version of uv
+        uses: astral-sh/setup-uv@v4
+        with:
+          version: "latest"
+          python-version: '3.10'
 
-      - name: Verify release version > latest PyPI version
-        id: verify_version
+      - name: Find latest release from PyPI
+        id: find_previous_release_branch
         run: |
-          RELEASE_VERSION="${{ github.event.inputs.release_version }}"
-          echo "Validated release version: ${RELEASE_VERSION}"
-
           # Get the latest version from PyPI using JSON API
           LATEST_PYPI_VERSION=$(curl -s https://pypi.org/pypi/skypilot/json | python -c "import sys, json; print(json.load(sys.stdin)['info']['version'])")
```

btw, we could use jq here, it might be easier. Not that important.
```suggestion
          LATEST_PYPI_VERSION=$(curl -s https://pypi.org/pypi/skypilot/json | jq -r .info.version)
```